### DISCUSSION TITLE
Alexa Matter - detecting only 1 device

### DISCUSSION THREAD

**Discussion by storchalberto** [arendst/Tasmota#19864] [Q&A] [answered]
I have a tasmota device with 5 relays, matter enabled, but alexa recognize only the first device and do not use the name inserted in the matter configuration page but use the Friendly Name 1
![Alexa_Tasmota_Matter](https://github.com/arendst/Tasmota/assets/32262123/05fce249-e5c3-4412-90ea-ba0addf56605)

**Comment by s-hadinger** [comment, marked as answer]
This is an Alexa limitation

**Reply by storchalberto** [reply]
I have also tested a configuration with a dht11 thermometer and one relays and ALEXA recognize only the temperature metering (the first device in the matter's list) and not the other elements. Have you a test program to verify that the device respond correctly in the matter protocol?

**Reply by s-hadinger** [reply]
Alexa seems to consider only endpoint 1. 

Tasmota works well with chip-tool, the problem is not on Tasmota side

**Reply by storchalberto** [reply]
Ok, i have tested the 5 relays tasmota device using Home Assistant. Here the 5 relays are recognized and controlled but the name displayed for all the relays is the Tasmota friendly name 1 
![immagine](https://github.com/arendst/Tasmota/assets/32262123/0cd012b6-eab7-43b4-bc43-fc689b5c2bef)

**Reply by s-hadinger** [reply]
That's a problem on HA side. The name is only working in bridge mode (which is the default). Maybe HA has no specific support for the bridge mode.

**Comment by lboue** [comment]
@storchalberto 
Same issue for me with HA  https://github.com/home-assistant/core/issues/102161.
It works fine with iOS.

**Comment by storchalberto** [comment]
So i understand that the "name" present in the "Local sensors and device" of the configuration page of Tasmota is used by Matter only in the Tasmota device is used as a bridge (maybe usefull to remove the text input in the matter configuration page for the devices not bridged and to document this characteristics). But for  functions of the Tasmota device itself (not bridged), i think that the name used are the "Other Parameters->Friendly name". The problem is that the name shown in the HA page for the multiple relays present in the Tasmota board is only the Friendly name 1.

**Reply by s-hadinger** [reply]
The fact that there are names on endpoint is only available in bridge mode, is a limitation of the Matter protocol. 

The fact that HA doesn't read the endpoint name is most certainly a limitation of HA. Apple and Google do report the endpoint name in their UI
